### PR TITLE
Add bug reporter as a feature

### LINKED ISSUE #69: Use bug reporter as a feature

### PR DESCRIPTION
**Pull request by saurabhdaalia**
Added Functionality to extract Bug Reporter as specified in the issue #69 
I have added it as a feature to the bug model. But there is no change in the accuracy of bug model and regression model.

Fixes #69.

### CODE REVIEW

**Comment by rhcu** [comment]
Hi @saurabhdaalia. Probably this could be helpful: currently you have Travis check failing because of code style that is checked with `flake8` - whitespaces, blank lines, isort. You can see the full log of Travis here: https://travis-ci.org/mozilla/bugbug/builds/485128380?utm_source=github_status&utm_medium=notification.

File: bugbug/bug_features.py [hunk 1]
```diff
@@ -135,6 +135,21 @@ def __call__(self, bug):
         return any(bug['creator_detail']['email'].endswith(domain) for domain in ['@mozilla.com', '@mozilla.org'])
 
 
+class bug_reporter(object):
+    def __call__(self, bug):
+        reporter_name = bug['creator_detail']['real_name']
```
Comment (marco-c) [inline bugbug/bug_features.py]: We should just return the reporter email, it's easier.

File: bugbug/bug_features.py [hunk 2]
```diff
@@ -162,6 +177,7 @@ def __call__(self, bug):
         return sum((commit['types'] for commit in bug['commits']), [])
 
 
+
```
Comment (marco-c) [inline bugbug/bug_features.py]: Unneeded whitespace change

File: bugbug/bug_features.py [hunk 3]
```diff
@@ -262,5 +278,5 @@ def transform(self, bugs):
             }
 
             results.append(result)
-
+
```
Comment (marco-c) [inline bugbug/bug_features.py]: Same here

File: bugbug/model.py [hunk 4]
```diff
@@ -75,6 +75,7 @@ def bugs():
             shap_sums = np.abs(shap_values).sum(0)
             rel_shap_sums = shap_sums / shap_sums.sum()
             indices = np.argsort(rel_shap_sums)[::-1][:20]
+
```
Comment (marco-c) [inline bugbug/model.py]: Same here

File: bugbug/models/bug.py [hunk 5]
```diff
@@ -33,6 +33,7 @@ def __init__(self, lemmatization=False):
             bug_features.patches(),
             bug_features.landings(),
             bug_features.title(),
+            bug_features.bug_reporter()
```
Comment (marco-c) [inline bugbug/models/bug.py]: Don't add it to this model if it doesn't change the result

File: bugbug/models/component.py [hunk 6]
```diff
@@ -34,6 +34,7 @@ def __init__(self, lemmatization=False):
             bug_features.patches(),
             bug_features.landings(),
             bug_features.title(),
+
```
Comment (marco-c) [inline bugbug/models/component.py]: Unneeded whitespace change

**Comment by marco-c** [comment]
Can you check if there is any change in the accuracy of the `tracking` model?

**Comment by saurabhdaalia** [comment]
@rhcu Thank you so much for the suggestions. 
@marco-c  why is build still failing?

**Comment by marco-c** [comment]
>  @marco-c why is build still failing?

The build failure was unrelated to your changes. Can you rebase on top of master?

And don't forget about:
> Can you check if there is any change in the accuracy of the tracking model?

**Comment by saurabhdaalia** [comment]
@marco-c sure I will do that. 
Yes I just checked, the accuracy does seem to improve 
Without Bug_Reporter Feature : CV Accuracy: f0.8668139777728052 (+/- 0.032730845922501375)
With Bug_Reporter Feature : CV Accuracy: f0.8680362870142464 (+/- 0.03451266444963794)

**Comment by saurabhdaalia** [comment]
@marco-c  Should I include bug reporter feature into tracking model?

**Comment by marco-c** [comment]
Yes, thanks! Then rebase on top of master to resolve the conflicts.

File: bugbug/models/component.py (hunk 6 above)
Comment (marco-c) [reply bugbug/models/component.py]: There is still an unneeded newline here

Comment (saurabhdaalia) [reply bugbug/models/component.py]: @marco-c , My bad

File: bugbug/models/component.py [hunk 7]
```diff
@@ -33,7 +33,7 @@ def __init__(self, lemmatization=False):
             bug_features.whiteboard(),
             bug_features.patches(),
             bug_features.landings(),
-            bug_features.title(),
+            bug_features.title()
```
Comment (marco-c) [inline bugbug/models/component.py]: You removed a comma :smile: 
Could you put it back?

File: tests/test_bug_features.py [hunk 8]
```diff
@@ -117,3 +122,5 @@ def test_comment_count():
 
 def test_comment_length():
     read('comment_length.json', comment_length, [566, 5291])
+
```
Comment (marco-c) [inline tests/test_bug_features.py]: Now you have two empty lines here :rofl:

Comment (saurabhdaalia) [reply tests/test_bug_features.py]: I feel like an idiot :sob:

Comment (marco-c) [reply tests/test_bug_features.py]: No worries, it happens to everyone :P

**Pull request merged**